doc/nrf/libraries/others/flash_map_pm.rst

```diff
@@ -0,0 +1,55 @@
+.. _lib_flash_map_pm:
+
+Partition Manager flash map
+###########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Partition Manager flash map library provides a set of defines for reading flash partition information when using the :ref:`partition_manager` script.
+This is mandatory for providing a unified interface for flash partitions when Partition Manager is used.
+
+Overview
+********
+
+In Zephyr, :file:`flash_map.h` provides unified interface to code for reading the flash area data, such as offset, size, and device.
+By default, the header file gets this flash area information from :ref:`zephyr:dt-guide` (namely, :file:`zephyr.dts`).
+
+In the |NCS|, when Partition Manager is used, flash area information is not taken from Devicetree.
+Instead, it is taken from :file:`pm_config.h`.
+In this configuration, :file:`flash_map_pm.h` acts as Partition Manager's backend for the flash area interface.
+
+Translation of partition labels
+===============================
+
+The library also provides a translation of partition labels used in upstream Zephyr.
+This is because Devicetree provides only one label for storage, which makes it impossible to enable two file systems at the same time.
```

Comment:
I don't think this is really true. The fixed partitions defined via devicetree are arbitrary. Upstream zephyr boards only define one such partition for storage by convention, but that's a historical circumstance, not a strict limitation. The C code which consumes the devicetree partitions can do whatever it wants, including enabling two file systems. So if the limitation exists, it is not in devicetree itself.